docs/deployment_kubernetes_docker.md

```diff
@@ -0,0 +1,199 @@
+contains the prediction service defined above, python code dependencies and PyPi
+dependencies, and the trained pet classification model. :
+
+```python
+# 1) import the custom BentoService defined above
+from pet_classification import PetClassification
+
+# 2) `pack` it with required artifacts
+service = PetClassification()
+service.pack('pet_classifier', learn)
+
+# 3) save your BentoService
+service.save()
+```
+
+### Validate the saved bundle by using prediction request with sample data
+
+BentoML automatically process the incoming data into required data format defined in the
+API. For the pet classifier BentoService defined above, incoming data will transform to
+fastai `ImageData` object.
+
+Make prediction request with sample data using BentoML CLI:
+
+```bash
+# Replace PATH_TO_TEST_IMAGE_FILE with one of the image from {path_img}
+# An example path: /Users/user_name/.fastai/data/oxford-iiit-pet/images/shiba_inu_122.jpg
+
+bentoml run PetClassification:latest predict --input=PATH_TO_TEST_IMAGE_FILE
+```
+
+## Deploy to Kubernetes
+
+### Build and push API server image
+
+BentoService SaveBundle directory structured as a docker context, that can be used to
+build an API server docker image.
+
+Replace `docker_username` with Docker Hub username:
+
+```bash
+saved_path=$(bentoml get IrisClassifier:latest -q | jq -r ".uri.uri")
+
+docker build -t {docker_username}/pet-classifier .
+docker push {docker_username}/pet-classifier
+```
+
+### Apply deployment to Kubernetes
+
+The following code creates a kubernetes resources spec. Replace `{docker_username}`
+with Docker Hub username and save to a file called
+`pet-classifier.yaml`:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: pet-classifier
+  name: pet-classifier
+spec:
+  ports:
+  - name: predict
+    port: 5000
+    targetPort: 5000
+  selector:
+    app: pet-classifier
+  type: LoadBalancer
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: pet-classifier
+  name: pet-classifier
+spec:
+  selector:
+    matchLabels:
+      app: pet-classifier
+  template:
+    metadata:
+      labels:
+        app: pet-classifier
+    spec:
+      containers:
+      - image: {docker_username}/pet-classifier
+        name: pet-classifier
+        ports:
+        - containerPort: 5000
+```
+
+Use `kubectl` to apply spec to the kubernetes cluster.
+
+```bash
+kubectl apply -f pet-classifier.yaml
+```
+
+Check deployment status with `kubectl`:
+
+```bash
+kubectl get svc pet-classifier
+```
+
+### Send prediction request
+
+Replace `EXTERNAL_IP` with Kubernetes cluster IP. For this guide, to get external IP
+use `minikube` command:
+
+```bash
+minikube ip
+```
+
+Make prediction request with `curl`:
+
+```bash
+# Replace PATH_TO_TEST_IMAGE_FILE
+curl -i \
+    --request POST \
+    --header "Content-Type: multipart/form-data" \
+    -F "image=@PATH_TOTEST_IMAGE_FILE" \
+    EXTERNAL_IP:5000/predict
```

Comment:
just ues:
```$(minikube ip):500/predcit``` here? 

you can remove l 177-182, and add a line below explaining if not using minikube, you will replace it with the cluster IP